### PR TITLE
LIME-545 - Calculating activity history score

### PR DESCRIPTION
**Pull request by ElliotMurphyGDS**
### What changed

Added functionality that calculates activity history score

### CODE REVIEW

File: lambdas/fraudcheck/src/test/java/uk/gov/di/ipv/cri/fraud/api/service/ActivityHistoryScoreCalculatorTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        Integer monthsBetween =
+                Math.toIntExact(ChronoUnit.MONTHS.between(date3MonthsAgoAsYyyMM, YearMonth.now()));
+
+        int activityHistoryScore =
+                activityHistoryScoreCalculator.calculateActivityHistoryScore(monthsBetween);
+        assertEquals(0, activityHistoryScore);
+    }
+
+    @Test
+    void testActivityHistoryScore1() {
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/test/java/uk/gov/di/ipv/cri/fraud/api/service/ActivityHistoryScoreCalculatorTest.java:35]: We should have a test in here for null as well for exception scenarios

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java [hunk 2]
```diff
@@ -263,27 +227,56 @@ private Integer getDecisionScore(List<DecisionElement> decisionElements, String
...
-        LocalDate date = LocalDate.parse(fieldDate, DateTimeFormatter.ofPattern("yyyyMM"));
-        Duration diff = Duration.between(date.atStartOfDay(), LocalDate.now().atStartOfDay());
-        long diffDays = diff.toDays();
-        long dateInMonths = diffDays / 12;
-        String approxDateInYears = "No value present";
+    private String logRecordAge(String fieldDate, String fieldName) {
+        try {
+            LocalDate date = LocalDate.parse(fieldDate, DateTimeFormatter.ofPattern("yyyyMM"));
+            Duration diff = Duration.between(date.atStartOfDay(), LocalDate.now().atStartOfDay());
+            long diffDays = diff.toDays();
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:234]: This a bit confusing do we still have days can we try and keep this and the logic in method below similar

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java [hunk 3]
```diff
@@ -263,27 +227,56 @@ private Integer getDecisionScore(List<DecisionElement> decisionElements, String
...
+            return approxDateInYears;
+        } catch (Exception e) {
+            LOGGER.info("Invalid value in reponse for {}", fieldName);
+            return null;
         }
+    }
+
+    private Integer calculateOldestDateInMonths(Integer... recordDateValues) {
+
+        List<Integer> dataCounts = new ArrayList<>();
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:261]: Does this need a try catch around it? We dont want to fail the whole fraud check on a bad parse

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java [hunk 4]
```diff
@@ -263,27 +227,56 @@ private Integer getDecisionScore(List<DecisionElement> decisionElements, String
...
 
-        LOGGER.info(
-                "Logging activity history score related value in response {} {}",
-                fieldName,
-                approxDateInYears);
-        return approxDateInYears;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMM");
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:263]: Do we have unit tests covering all the discussed scenarios for this

Comment (ElliotMurphyGDS) [reply lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:263]: Yes, see identitiyVerificationResponseMapperTest

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java (hunk 3 above)
Comment (ElliotMurphyGDS) [reply lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:261]: Yes...not sure how my try catches got removed

**Review by MMcC-GDS — DISMISSED** [review]

**Review by MMcC-GDS — DISMISSED** [review]

**Review by gds-jbentham — APPROVED** [review]

**Pull request merged**